code/modules/mob/living/silicon/robot/robot_vr.dm

```diff
@@ -30,3 +30,42 @@
 			overlays += "[module_sprites[icontype]]-sleeper_g"
 		if(sleeper_r == 1)
 			overlays += "[module_sprites[icontype]]-sleeper_r"
+
+/mob/living/silicon/robot/Move(a, b, flag)
+
+	. = ..()
+
+	if(module)
+		if(module.type == /obj/item/weapon/robot_module/scrubpup)
+			var/turf/tile = loc
+			if(isturf(tile))
```

Comment:
Okay.  I think here what we want to do is call  `T.clean(src, src)` which will handle the turf cleaning part for you.  It will also use up some water, which I think is a good idea.    Just need to add a little bit of extra code allowing the dogborg to drink from a sink to refill their reagents with water!